docs/contributing-to-registry.md

```diff
@@ -0,0 +1,66 @@
+
+# Contributing to Bicep public registry
+
+Currently, we only accept contributions from Microsoft employees. The guide was created to help teams within Microsoft with the development of Bicep public registry modules. External customers can propose a new module by opening an [issue]() in the [Azure/azure-quickstart-templates](https://github.com/Azure/azure-quickstart-templates/tree/master/modules) repository.
+
+## Prerequisite
+- Create a fork of the [Azure/azure-quickstart-templates](https://github.com/Azure/azure-quickstart-templates/tree/master/modules) repository and clone the fork to your local machine.
+<!-- TODO: Add link to Nuget once the tool is published -->
+- Install the [bicep-registry-module]() .NET tool by running:
+  - `dotnet tool install -g bicep-registry-module`
+
+## Creating a new module
+
+### Making a proposal for new module
+<!-- TODO: create an issue template in the quickstart repo -->
+Before creating a new module, you must fill out this [issue template](https://github.com/Azure/azure-quickstart-templates/issues/new) to make a proposal. Once the proposal is approved, proceed with the following steps.
+
+### Creating a directory for the new module
+<!-- TODO: need to discuss the pattern of the module path -->
+Add a new directory under the `modules` folder in your local azure-quickstart-templates repository with the path following the pattern `<ResourceProviderName>/<ModuleName>/<ModuleMajorVersion>.<ModuleMinorVersion>`. For examples: 
+- `Microsoft.Storage/storageAccounts/0.9`
+- `Microsoft.Web/webApplications/1.0`
+
+### Generating module files
+Open a terminal and navigate to the newly created folder. From there, run the following command to generate the required files for the Bicep public registry module:
+```
+bicep-registry-module generate
+```
+
+You should be able to see these files created in the module folder:
+- `metadata.json`
+  - The module manifest file that contains metadata you must provide, such as `description`.
+- `main.bicep`
+  - An empty Bicep file to be updated.
+- `azuredeploy.json`
+  - The ARM template file generated from `main.bicep`. This is the file that will be published to the Bicep public registry. You should not modify this file.
+- `azuredeploy.parameters.json`
+  - The ARM template parameters file for `azuredeploy.json`. The file is generated for running template deployment validation. You must provide values for the required parameters in the file.
+- `README.md`
+  - The README file generated based on the contents of `metadata.json` and `main.bicep`. Do not update this file manually.
+- `version.json`
+  - The version file, together with the `azuredeploy.json` file, are used to calculate the patch version of the module automatically when publishing the module to the Bicep public module registry. You should not edit this contents of the file.
+
+### Authoring module files
```

Comment:
Yeah, we should be able to. In fact, it is much easier to implement the functionality in Bicep CLI since we have access to the semantic model. In the registry module tool we had to parse the ARM template file to get the parameters. Once we implemented parameter generation in Bicep CLI, we should change the registry module tool to invoke Bicep CLI to generate parameters.